controllers/glanceapi_controller.go

```diff
@@ -1000,3 +1034,122 @@ func (r *GlanceAPIReconciler) ensureDeletedEndpoints(
 	}
 	return ctrl.Result{}, err
 }
+
+func (r *GlanceAPIReconciler) ensureImageCacheJob(
+	ctx context.Context,
+	h *helper.Helper,
+	instance *glancev1.GlanceAPI,
+	serviceLabels map[string]string,
+	serviceAnnotations map[string]string,
+	cjType glance.CronJobType,
+) (ctrl.Result, error) {
+
+	var err error
+	var ctrlResult ctrl.Result
+
+	command := glance.GlanceCacheCleaner
+	schedule := instance.Spec.ImageCache.CleanerScheduler
+
+	if cjType == glance.CachePruner {
+		command = glance.GlanceCachePruner
+		schedule = instance.Spec.ImageCache.PrunerScheduler
+	}
+	cachePVCs, _ := GetPvcListWithLabel(ctx, h, instance.Namespace, serviceLabels)
+	for _, vc := range cachePVCs.Items {
+		var pvcName string = vc.GetName()
+		cacheAnnotations := vc.GetAnnotations()
+		if _, ok := cacheAnnotations["image-cache"]; ok {
+			cronSpec := glance.CronJobSpec{
+				Name:        fmt.Sprintf("%s-%s", pvcName, cjType),
+				PvcClaim:    &pvcName,
+				Command:     command,
+				CjType:      cjType,
+				Schedule:    schedule,
+				Labels:      serviceLabels,
+				Annotations: serviceAnnotations,
+			}
+			cronjobDef := glanceapi.ImageCacheJob(
+				instance,
+				cronSpec,
+			)
+			imageCacheCronJob := cronjob.NewCronJob(
+				cronjobDef,
+				5*time.Second,
+			)
+			ctrlResult, err := imageCacheCronJob.CreateOrPatch(ctx, h)
+			if err != nil {
+				return ctrlResult, err
+			}
+		}
+	}
+	return ctrlResult, err
+}
+
+// deleteImageCacheJob - delete the ImageCache cronJobs associated to a given
+// GlanceAPI
+func (r *GlanceAPIReconciler) deleteImageCacheJob(
+	ctx context.Context,
+	h *helper.Helper,
+	instance *glancev1.GlanceAPI,
+	serviceLabels map[string]string,
+) (ctrl.Result, error) {
+	var err error
+	var ctrlResult ctrl.Result
+	// Get the PVCs using labelSelector (only the PVCs associated to the current
+	// GlanceAPI are retrieved)
+	cachePVCs, _ := GetPvcListWithLabel(ctx, h, instance.Namespace, serviceLabels)
+
+	// For each PVC that present the image-cache annotation, check if there's an
+	// associated POD (pvc and pod shares the same StatefulSet templating mechanism)
+	for _, vc := range cachePVCs.Items {
+		cacheAnnotations := vc.GetAnnotations()
+		if _, ok := cacheAnnotations["image-cache"]; ok {
+			var pvcName string = vc.GetName()
+			// Get the pod (by name) associated to the current pvc
+			var pod corev1.Pod
+			if err := r.Client.Get(ctx, types.NamespacedName{
+				Name:      strings.TrimPrefix(pvcName, "glance-cache-"),
+				Namespace: instance.Namespace,
+			}, &pod); err != nil && k8s_errors.IsNotFound(err) {
```

Comment:
If the `err` is not `nil` and isn't a `NotFound`, could we mistakenly miss catching and returning it here?  i.e. the loop would just iterate to the next `vc` and the old `err` would be lost?  Maybe it's just a pedantic concern and doesn't really matter.

Reply:
yeah, basically here we try to get a `Pod` that matches w/ the `Pvc` name (excluding the prefix introduced by the `StatefulSet` templating), and if the pod is not found, we need to perform a delete on the `cronJobs` that are still there.
That means that `err` doesn't matter here, as we want to just move to the next element and be silent in case a `Pod` or a `Cronjob` is not found (that's the only condition where we need the current value of `err`)